beets/autotag/match.py

```diff
@@ -294,23 +313,24 @@ def tag_album(
                         Proposal(list(candidates.values()), rec),
                     )
 
-        # Search terms.
-        if not (search_artist and search_album):
-            # No explicit search terms -- use current metadata.
-            search_artist, search_album = cur_artist, cur_album
-        log.debug("Search terms: {} - {}", search_artist, search_album)
+        # Manually provided search terms or fallbacks.
+        _search_artist, _search_album = _parse_search_terms_with_fallbacks(
+            (search_artist, cur_artist),
+            (search_album, cur_album),
+        )
```

Comment:
Based on my comment above, we don't need to handle `None`s any more, so this can be simplified to

```py
if not search_artist and not search_album:
    search_artist, search_album = cur_artist, cur_album
```